config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/TrustedCertEntityBuilder.java

```diff
@@ -0,0 +1,149 @@
+        final String id = idGenerator.generate();
+        final Element trustedCertElem = createElementWithAttributesAndChildren(
+                document,
+                TRUSTED_CERT,
+                ImmutableMap.of(ATTRIBUTE_ID, id),
+                createElementWithTextContent(document, NAME, name),
+                buildCertData(name, trustedCert, certificateFiles)
+        );
+        buildAndAppendPropertiesElement(trustedCert.createProperties(), document, trustedCertElem);
+
+        return new Entity(TRUSTED_CERT_TYPE, name, id, trustedCertElem);
+    }
+
+    private Element buildCertData(String name, TrustedCert trustedCert, Map<String, File> certificateFiles) {
+        if (name.startsWith("https://")) {
+            return buildCertDataFromUrl(name);
+        } else if (certificateFiles.get(name) != null) {
+            return buildCertDataFromFile(certificateFiles.get(name));
+        } else if (trustedCert.getCertificateData() != null) {
+            final CertificateData certData = trustedCert.getCertificateData();
+            return createCertDataElementFromCert(
+                    certData.getIssuerName(),
+                    certData.getSerialNumber(),
+                    certData.getSubjectName(),
+                    certData.getEncodedData()
+            );
+        } else {
+            throw new EntityBuilderException("Trusted Cert must be loaded from a specified url," +
+                    " or from a certificate file that has the same name as the Trusted Cert.");
+        }
+    }
+
+    private Element buildCertDataFromFile(File certFileLocation) {
+        if (!certFileLocation.exists()) {
+            throw new EntityBuilderException("The certificate file location is not specified.");
+        }
+
+        try (FileInputStream is = new FileInputStream(certFileLocation)) {
+            CertificateFactory certFact = CertificateFactory.getInstance("X.509");
+            X509Certificate cert = (X509Certificate) certFact.generateCertificate(is);
+            return createCertDataElementFromCert(
+                    cert.getIssuerDN().getName(),
+                    cert.getSerialNumber(),
+                    cert.getSubjectDN().getName(),
+                    Base64.getEncoder().encodeToString(cert.getEncoded())
+            );
+        } catch (IOException e) {
+            throw new EntityBuilderException("The certificate file location specified does not exist.");
+        } catch (CertificateException e) {
+            throw new EntityBuilderException("Error generating certificate: ", e);
+        }
+    }
+
+    private Element buildCertDataFromUrl(String name) {
+        try {
+            final URL url = new URL(name);
+            int port = url.getPort() == -1 ? url.getDefaultPort() : url.getPort();
+            try (SSLSocket socket = (SSLSocket) acceptAllSocketFactory.createSocket(url.getHost(), port)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1141))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=68&issues=AWX4rQKE2IDoRcZ5VlPr&open=AWX4rQKE2IDoRcZ5VlPr)

[ISSUE_KEY]: AWX4rQKE2IDoRcZ5VlPr